From: Victor Seva <linuxmaniac@torreviejawireless.org>
Date: Tue, 20 Sep 2016 17:49:19 +0200
Subject: sca: fix core when HDR_CSEQ not parsed

> Program terminated with signal SIGSEGV, Segmentation fault.
> #0  0x00007f1820227bae in sca_call_info_update (msg=0x7f1828635930, p1=0x0, p2=0x0, p3=0x7f1828635930 "\001") at sca_call_info.c:1845
> 1845			LM_DBG("BUG: sca module does not support Call-Info headers "
> (gdb) p msg->cseq
> $1 = (struct hdr_field *) 0x0
> (gdb) p msg->first_line.u.request->method
> $2 = {
>  s = 0xa33540 <buf> "SUBSCRIBE sip:30@127.0.0.1:5060 SIP/2.0\r\nVia: SIP/2.0/UDP 172.17.0.2:5061;branch=z9hG4bK-2884-1-0\r\nFrom: sipp <sip:sipp@172.17.0.2:5061>;tag=2884SIPpTag001\r\nTo: sut <sip:30@127.0.0.1:5060>\r\nCall-ID: 1"..., len = 9}
---
 modules/sca/sca_call_info.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/sca/sca_call_info.c b/modules/sca/sca_call_info.c
index 1c74f6f..a5854cc 100644
--- a/modules/sca/sca_call_info.c
+++ b/modules/sca/sca_call_info.c
@@ -1901,6 +1901,12 @@ sca_call_info_update( sip_msg_t *msg, char *p1, char *p2, char *p3 )
 
     if ( p1 != NULL ) {
 	if ( get_int_fparam( &update_mask, msg, (fparam_t *)p1 ) < 0 ) {
+        if(msg->cseq==NULL && ((parse_headers(msg, HDR_CSEQ_F, 0)==-1) ||
+           (msg->cseq==NULL)))
+        {
+           LM_ERR("no CSEQ header\n");
+           return (1);
+        }
 	    LM_ERR( "sca_call_info_update: argument 1: bad value "
 		    "(integer expected)" );
 	    return( -1 );
